src/tree/updater_quantile_hist.cc

```diff
@@ -1138,9 +1138,9 @@ void QuantileHistMaker::Builder<GradientSumT>::FindSplitConditions(
     // split_cond = -1 indicates that split_pt is less than all known cut points
     CHECK_LT(upper_bound,
              static_cast<uint32_t>(std::numeric_limits<int32_t>::max()));
-    for (uint32_t i = lower_bound; i < upper_bound; ++i) {
-      if (split_pt == gmat.cut.Values()[i]) {
-        split_cond = static_cast<int32_t>(i);
+    for (uint32_t bound = lower_bound; bound < upper_bound; ++bound) {
+      if (split_pt == gmat.cut.Values()[bound]) {
+        split_cond = static_cast<int32_t>(bound);
```

Comment:
Why are we renaming the loop variable here? The name `i` is commonly used as a loop variable, and there's no need to rename it here. I also don't think the renaming fits the purpose of this pull request (reduce compiler warning).

Reply:
It does cause a warning because it is inside another foor loop which has declared another `i`as iterator

Reply:
Got it. We can keep this one then.